nativelib/src/main/resources/scala-native/gc/none/gc.c

```diff
@@ -9,19 +10,58 @@
 // Dummy GC that maps chunks of 4GB and allocates but never frees.
 
 // Map 4GB
-#define CHUNK (4 * 1024 * 1024 * 1024L)
+#define DEFAULT_CHUNK (4 * 1024 * 1024 * 1024L)
 // Allow read and write
 #define DUMMY_GC_PROT (PROT_READ | PROT_WRITE)
 // Map private anonymous memory, and prevent from reserving swap
-#define DUMMY_GC_FLAGS (MAP_NORESERVE | MAP_PRIVATE | MAP_ANONYMOUS)
+#define DEFAULT_DUMMY_GC_FLAGS (MAP_NORESERVE | MAP_PRIVATE | MAP_ANONYMOUS)
 // Map anonymous memory (not a file)
 #define DUMMY_GC_FD -1
 #define DUMMY_GC_FD_OFFSET 0
 
 void *current = 0;
 void *end = 0;
 
+static size_t CHUNK = DEFAULT_CHUNK;
+static size_t DUMMY_GC_FLAGS = DEFAULT_DUMMY_GC_FLAGS;
+static size_t TO_NORMAL_MMAP = 1L;
+
+size_t set_prealloc() {
+    return (MAP_NORESERVE | MAP_PRIVATE | MAP_ANONYMOUS | MAP_POPULATE);
+}
+
+void Prealloc_Or_Default() {
+    if (TO_NORMAL_MMAP == 1L) { // Check if we have prealloc env varible or
+                                // execute default mmap settings
+        size_t PreallocSize = Parse_Env_Or_Default("GC_INITIAL_HEAP_SIZE", 0);
+
+        if (PreallocSize == 0) { // no prealloc settings.
+            TO_NORMAL_MMAP = 0L;
+
+        } else { // config prealloc settings and the flag to reset the mmap
+                 // settings the next iteration.
+            if (PreallocSize <= getMemorySize()) {
+                CHUNK = PreallocSize;
+                DUMMY_GC_FLAGS = set_prealloc();
+                TO_NORMAL_MMAP = 2L;
+
+            } else if (PreallocSize > getMemorySize() && getMemorySize() > 0) {
+
+                CHUNK = getMemorySize();
```

Comment:
...and here once more.

Reply:
The intention was deliberate, because using getAvailSize()(the previous version of getMemorySize function) was getting the current available memory. Using getMemorySize() gives a static answer (it gets the total amount of memory). This is not critical, we only call this three times as you pointed, but for the sake of good code style i will factor it to use only one getMemorySize() call.

Reply:
OK good. Yes, I couldn't tell if there was some OS side-effect or some temporal consideration, but if not it is easier to grok if changed like so and I agree also better coding style.